src/vs/workbench/contrib/terminal/browser/chatTerminalCommandMirror.ts

```diff
@@ -0,0 +1,98 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from '../../../../base/common/lifecycle.js';
+import type { ITerminalCommand } from '../../../../platform/terminal/common/capabilities/capabilities.js';
+import { ITerminalService, type IDetachedTerminalInstance } from './terminal.js';
+import { DetachedProcessInfo } from './detachedTerminal.js';
+import { XtermTerminal } from './xterm/xtermTerminal.js';
+import { TERMINAL_BACKGROUND_COLOR } from '../common/terminalColorRegistry.js';
+import { PANEL_BACKGROUND } from '../../../common/theme.js';
+
+interface IDetachedTerminalCommandMirror {
+	attach(container: HTMLElement): Promise<void>;
+	renderCommand(): Promise<{ lineCount?: number } | undefined>;
+}
+
+/**
+ * Mirrors a terminal command's output into a detached terminal instance.
+ * Used in the chat terminal tool progress part to show command output for example.
+ */
+export class DetachedTerminalCommandMirror extends Disposable implements IDetachedTerminalCommandMirror {
```

Comment:
We want some unit tests for this, they'll be a lot more important in the next PR though. Basically they should write stuff to `XtermTerminal`, pass in the command and verify they both serialize to the same text. For the streaming PR we want it to check as things are progressing.